### Making a prediction with multiple inputs

In [2]:
weights = [0.1, 0.2, 0]

#performing weighted sum of inputs
def w_sum(a,b):
    assert(len(a) == len(b))
    output = 0
    
    for i in range(len(a)):
        output += (a[i] * b[i])
        
    return output


def neural_network(input, weights):
    pred = w_sum(input, weights)
    return pred

This dataset is the current status at the beginning of each game for the first four games in a season:
- toes = current average number of toes per player
- wirec = current games won (present)
- nfans = fan count (in millions)

In [4]:
toes = [8.5, 9.5, 9.9, 9.0]
wirec = [0.65, 0.8, 0.8, 0.9]
nfans = [1.2, 1.3, 0.5, 1.0]

input = [toes[0], wirec[0], nfans[0]] #gives you one row of data for the first game
pred = neural_network(input, weights)

print(pred)

0.9800000000000001


In [6]:
import numpy as np

In [7]:
weights = np.array([0.1, 0.2, 0])
def neural_network(input, weights):
    pred = input.dot(weights)
    return pred

toes = np.array([8.5, 9.5, 9.9, 9.0])
wlrec = np.array([0.65, 0.8, 0.8, 0.9])
nfans = np.array([1.2, 1.3, 0.5, 1.0])

input = np.array([toes[0], wlrec[0], nfans[0]])
pred = neural_network(input, weights)
print(pred)

0.9800000000000001


### Making a prediction with multiple outputs

In [10]:
#performing elementwise multiplication
def ele_mul(number, vector):
    output = [0,0,0]
    assert(len(output) == len(vector))
    for i in range(len(vector)):
        output[i] = number * vector[i]
        
    return output

weights = [0.3, 0.2, 0.9]
def neural_network(input, weights):
    pred = ele_mul(input,weights)
    return pred

wlrec = [0.65, 0.8, 0.8, 0.9]
input = wlrec[0]
pred = neural_network(input, weights)
print(pred)

[0.195, 0.13, 0.5850000000000001]


predicting that 19.5% of the baseball team players are hurt, 13% chance that the team won, and 58.5% chance that the players are sad given the win lose record input.

### Predicting with multiple inputs and outputs

In [12]:
#performing weighted sum of inputs
def w_sum(a,b):
    assert(len(a) == len(b))
    output = 0
    
    for i in range(len(a)):
        output += (a[i] * b[i])
        
    return output

#Performing separate weighted sums
#vector matrix multiplication essentially
def vect_mat_mul(vect, matrix):
    assert(len(vect) == len(matrix))
    output = [0,0,0]
    
    for i in range(len(vect)):
        output[i] = w_sum(vect, matrix[i])
        
    return output

          # toes # win # fans
weights = [[0.1, 0.1, -0.3], # hurt?
           [0.1, 0.2, 0.0], # win ?
           [0.0, 1.3, 0.1]] # sad?
def neural_network(input, weights):
    pred = vect_mat_mul(input, weights)
    return pred

toes = [8.5, 9.5, 9.9, 9.0]
wlrec = [0.65, 0.8, 0.8, 0.9]
nfans = [1.2, 1.3, 0.5, 1.0]

input = [toes[0], wlrec[0], nfans[0]]
pred = neural_network(input, weights)
print(pred)

[0.555, 0.9800000000000001, 0.9650000000000001]


### Predicting on Predictions

In [13]:
        # toes # win # fans
ih_wgt = [[0.1, 0.2, -0.1], #hid[0]
          [-0.1,0.1, 0.9], #hid[1]
          [0.1, 0.4, 0.1]] #hid[2]
     # hid[0] #hid[1] #hid[2]
hp_wgt = [[0.3, 1.1,-0.3], #hurt?
          [0.1, 0.2, 0.0], #win?
          [0.0, 1.3, 0.1]] #sad?

weights = [ih_wgt, hp_wgt]

def neural_network(input, weights):
    hid = vect_mat_mul(input, weights[0])
    pred = vect_mat_mul(hid, weights[1])
    return pred

toes = [8.5, 9.5, 9.9, 9.0]
wlrec = [0.65, 0.8, 0.8, 0.9]
nfans = [1.2, 1.3, 0.5, 1.0]

input = [toes[0], wlrec[0], nfans[0]]
pred = neural_network(input, weights)
print(pred)
#[hurt?, win?, sad?]

[0.21350000000000002, 0.14500000000000002, 0.5065]


In [17]:
        # toes # win # fans
ih_wgt = np.array([
        [0.1, 0.2, -0.1], #hid[0]
        [-0.1,0.1, 0.9], #hid[1]
        [0.1, 0.4, 0.1]]).T #hid[2]

     # hid[0] #hid[1] #hid[2]
hp_wgt = np.array([
        [0.3, 1.1,-0.3], #hurt?
        [0.1, 0.2, 0.0], #win?
        [0.0, 1.3, 0.1]]).T #sad?

weights = [ih_wgt, hp_wgt]

def neural_network(input, weights):
    hid = input.dot(weights[0])
    pred = hid.dot(weights[1])
    return pred

toes = np.array([8.5, 9.5, 9.9, 9.0])
wlrec = np.array([0.65, 0.8, 0.8, 0.9])
nfans = np.array([1.2, 1.3, 0.5, 1.0])

input = np.array([toes[0], wlrec[0], nfans[0]])
pred = neural_network(input, weights)
print(pred)
#[hurt?, win?, sad?]

[0.2135 0.145  0.5065]
